src/summersplash2022/Content.tsx

```diff
@@ -0,0 +1,413 @@
+  const blobSection1 = require('./img/section1Blob.png');
+  const Marius = require('./img/Marius.png');
+  const summerStudent1 = require('./img/Vilde summerstudent.png');
+  const summerStudent2 = require('./img/Olepetter summerstudent.png');
+  const summerStudent3 = require('./img/Nora summerstudent.png');
+
+  const [whichButtonSelected, setWhichButtonSelected] = useState('');
+
+  const forside = useRef<HTMLImageElement>(null);
+  const sommerjobbHandling = useRef<HTMLImageElement>(null);
+  const sommerjobbereMening = useRef<HTMLImageElement>(null);
+  const hvorforVariant = useRef<HTMLImageElement>(null);
+  const etterSoknad = useRef<HTMLImageElement>(null);
+  const sokSommerjobb = useRef<HTMLImageElement>(null);
+
+  const goToRef = (ref: React.RefObject<HTMLImageElement>) => {
+    ref?.current?.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  // Undefined for now. Need to be changed to 'internship' when index sends a prop.
+  if (selected === 'internship') {
+    return (
+      <>
+        <div className={style.scrollContainer}>
+          <section className={style.section1} ref={forside}>
+            <span className={style.searchNewVariants}>
+              <NewVariants />
+            </span>
+            <h4 className={style.h4Section1}>Søk innen 2. Oktober</h4>
+
+            <br />
+            {/* Navigation bar on the right side */}
+            <div className={style.navigationSlider}>
+              <button
+                className={style.navigationButton6}
+                onClick={() => goToRef(forside)}
+              >
+                <p className={style.tests}>Forside</p>
+              </button>
+              <button
+                className={style.navigationButton1}
+                onClick={() => goToRef(sommerjobbHandling)}
+              >
+                Hva går sommerjobben ut på
+              </button>
+              <button
+                className={style.navigationButton2}
+                onClick={() => goToRef(sommerjobbereMening)}
+              >
+                Hva mener årets sommerstudenter?
+              </button>
+              <button
+                className={style.navigationButton3}
+                onClick={() => goToRef(hvorforVariant)}
+              >
+                Hvorfor jobbe i Variant?
+              </button>
+              <button
+                className={style.navigationButton4}
+                onClick={() => goToRef(etterSoknad)}
+              >
+                Hva skjer etter søknadsfristen?
+              </button>
+              <button
+                className={style.navigationButton5}
+                onClick={() => goToRef(sokSommerjobb)}
+              >
+                Søk sommerjobb
+              </button>
+            </div>
+            <div className={style.underHeader}>
+              {/* middlertidig link. usikker på hvilken vi skal bruke */}
+              <a className={style.søkHer} href="https://www.variant.no/jobs">
```

Comment:
Lenker som brukes innbyrdes siden (om i dette tilfelle til en annen underside) har stor fordel av å brukes som "Next.js links" ([docs](https://nextjs.org/docs/api-reference/next/link)). Da vil ting prefetches og den vil prøve å navigere via klientside-kode direkte fremfor nytt sidelast. Så det vil da skje omtrent sånn:

1. Henter ting i bakgrunnen på klientsiden
2. Prøver å hente ut forskjellene
3. Replacer siden med det den henter ut

Resultatet er at navigasjonen vil føles kjappere og den slipper å hente ut like mye informasjon/gjøre like mye arbeid.

Eksempel:

```
<Link href="/jobs">
  <a className={style.søkHer}>Søk her</a>
</Link>
```

Reply:
Ville også ha vurdert å se på css-klasser som er på engelsk. Både for å matche resten av kodebasen, men også for man vet ikke helt alltid om slike verktøy har tatt høyde for særnorske tegn når de gjør parsing/resolving og alt som hører til.